src/main/kotlin/no/item/xp/plugin/GenerateTypeScriptTask.kt

```diff
@@ -1,27 +1,96 @@
 import org.gradle.api.tasks.TaskAction
+import org.w3c.dom.Document
+import java.lang.Exception
+import javax.xml.parsers.DocumentBuilder
 
+@Suppress("UNUSED_VARIABLE")
 open class GenerateTypeScriptTask @Inject constructor(private val extension: GenerateTypeScriptExtension) : DefaultTask() {
 
+  private val docBuilder: DocumentBuilder = DocumentBuilderFactory.newInstance().newDocumentBuilder()
+
     @TaskAction
     fun generateTypeScript() {
-      val files = findXmlFiles()
-      println("Found XML files: ")
-      println("--------------------")
-      files.forEach { println(it) }
-      // 2.send dem til en parser
-      // 3.send resultatet til en writer
-      // 4.exit
-    }
-
+      val files: Sequence<String> = findXmlFiles()
+      files.forEach { parseFile(it) }
+  }
   private fun findXmlFiles(): Sequence<String> {
     return File(extension.fileDir)
       .walk()
-      .filter { it.isFile && it.extension == "xml" }
+      .filter { it.isFile }
       .map { it.absolutePath }
   }
+
+  private fun parseFile(filePath: String) {
+    val xmlType: Either<Throwable, XmlType> = getXmlType(filePath)
+    var xmlTypeValue: XmlType? = null
+    when(xmlType){
+      is Either.Right -> {
+        xmlTypeValue = xmlType.b
+      }
+    }
+    if(xmlTypeValue!=null) {
+      val xmlFile: XmlFile = generateXmlFile(filePath, xmlTypeValue)
+      val document: Document = readXml(xmlFile)
+      parse(document)
+
+      val interfaceName: String? = GenerateInterfaceName(File(filePath))
+      val xml: Either<Throwable, Sequence<GeneratedField>> = parse(document)
+      /*if (interfaceName != null) {
+      TypeScriptWriter(interfaceName, xmlType, xml)
+    }*/
+    }
+  }
+
+  private fun getXmlType(filePath: String): Either<Throwable, XmlType> {
+    val child: Path = Paths.get(filePath).toAbsolutePath()
+    return returnIsChild(child)
+  }
+
+  private fun returnIsChild(child: Path): Either<Throwable, XmlType> {
+    for (pair: Pair<String, XmlType> in pairList) {
+      if (isChild(child, pair.first)) {
+        return Either.right(pair.second)
+      }
+    }
+    return Either.left(Exception("could not find XmlType for $child"))
+  }
+
+  private val pairList = listOf(
+    Pair("../site/content-types", XmlType.CONTENT_TYPE),
+    Pair("../site/layouts/", XmlType.LAYOUT),
+    Pair("../site/mixins/", XmlType.MIXIN),
+    Pair("../site/parts/", XmlType.PART),
+    Pair("../site/pages/", XmlType.PAGE),
+    Pair("../site/macros/", XmlType.MACRO),
+    Pair("../idprovider/", XmlType.ID_PROVIDER),
+    Pair("../site/", XmlType.SITE)
+  )
+
+  private fun isChild(child: Path, parentText: String): Boolean {
+    val parent: Path = Paths.get(parentText).toAbsolutePath()
+    return child.startsWith(parent)
+  }
+
+  private fun generateXmlFile(it: String, xmlType: XmlType): XmlFile {
+    return XmlFile(it, xmlType)
+  }
+
+  private fun readXml(it: XmlFile): Document {
+    val input = File(it.path)
+    return docBuilder.parse(input)
+  }
```

Comment:
I would just have this as a oneliner. It is simple enough. Maybe the name could be made even clearer. How about `getXmlDocumentByFile`?

And don't use `it` as a variable name. Usually it is good to use the whole type as the file name (but lower cased), or a part of the type. So either `xmlFile` or `file` or `xml`. Try to communicate as clearly as possible with your parameter names, and function names. 

```suggestion
  private fun getXmlDocumentByFile(xmlFile: XmlFile): Document =
    docBuilder.parse(File(xmlFile.path))
```